lib/Platform/ElasticSearchPlatform.php

```diff
@@ -409,6 +410,22 @@ private function connectToElastic(array $hosts) {
 		}
 	}
 
+	/**
+	 * @param array $hosts
+	 */
+	private function getBasicAuth(array $hosts) {
+		foreach ($hosts as $host) {
+			$parts = parse_url($host);
+			// We assume that the user/pass combination is the same in all the hosts 
+			// because ElasticSearch requires all hosts to have the same user/pass. So
+			// take the first entry which contains both user and pass.
+			if (isset($parts['user']) && isset($parts['pass'])) {
+				return [$parts['user'], $parts['pass']];
+			}
```

Comment:
**This is out of scope of this PR and isn't a blocker for merging:**

in a future PR it would be nice to also print a warning in case hosts URLs contain mismatching user/pass combinations. I think that this can be done reasonably easily with a combination of `array_map` and `array_uniq`.

But after thinking a bit we may as well just want to add explicit user/pass fields to the configuration page anyway.

Reply:
> But after thinking a bit, we may just want to add explicit user/pass fields to the configuration page at some point anyway.

Totally agree, that would be the cleanest solution but this one here is the backward compatibility fix :smile: 

Regarding the pass/user combo: I think logging could be nice. But I don't really understand your code suggestion. When leaving out either the username **OR** the password, AFAIK the basic auth would be skipped under the hood anyways, see [here](https://github.com/elastic/elasticsearch-php/blob/b3af4309bc73fa3b2a1e19f04f9f201916c11417/src/ClientBuilder.php#L379). So in both cases we would end up with a http client, where the auth header won't be set.


**A word to logging ...**
For the future it could also be nice to use the embedded client logging functionallity and registering a PSR-3 logger via [`setLogger`](https://github.com/elastic/elasticsearch-php/blob/b3af4309bc73fa3b2a1e19f04f9f201916c11417/src/ClientBuilder.php#L197). Think this would give a lot of `information` logs but could be helpful when debugging.

Reply:
Good point with the `elasticsearch` client SDK HTTP Basic Auth limitation. I haven't checked in ElasticSearch though, so I'm not sure if it's a bug of this client implementation of if it's in line with the behavior of ElasticSearch.

Reply:
Agree :+1: I guess they're modelling the ES behaviour. AFAIK the simple basic auth in ES always consists of a username (default `elastic`) and a password. So I think when just using the default we should be fine here.